src/algebra/group_power/basic.lean

```diff
@@ -220,6 +221,22 @@ by rw [pow_bit0', neg_mul_neg, pow_bit0']
 @[simp] theorem neg_pow_bit1 [ring R] (a : R) (n : ℕ) : (- a) ^ (bit1 n) = - a ^ (bit1 n) :=
 by simp only [bit1, pow_succ, neg_pow_bit0, neg_mul_eq_neg_mul]
 
+theorem iterate_mul_right (a : M) : ∀ n, (* a)^[n] = (* a^n)
```

Comment:
To be honest, I do not follow much the notation or the proof, but I think that there might be spaces missing aroung `^`?
```suggestion
theorem iterate_mul_right (a : M) : ∀ n, (* a) ^ [n] = (* a ^ n)
```

Reply:
Nope, this notation is correct as is:

https://github.com/leanprover-community/mathlib/blob/a3d951bc6ee53bcdbcea4080108579fcc62eb5a5/src/logic/function/iterate.lean#L55

https://github.com/leanprover-community/lean/blob/5a5c139af3e9b2514c80e4670546d7fbd04ba20d/library/init/data/nat/lemmas.lean#L792